unfetter-discover/unfetter-ui
# Issue 994 3

**Requires** a stix PR https://github.com/unfetter-discover/stix/pull/14
supports unfetter-discover/unfetter#994

## problem
need to create an assessment using a given question set from a users assessment baseline

## changes
* this allows the user to select a given baseline, and score against its capabilities
* added more error handling on observables, and display to page
* added error action to ngrx assess beta store
* used rxjs 5.5+ pipeable operators

## test
* ensure you have a baseline created w/ some capabilities
* go to assess beta wizard
* select assessment type include a 'capability'
* click next
* score the capabilities
* save the assessment, notice the summary and results page are mostly unfinished at the moment (will implement next)

* next test the error state
* visit an inprogress wizard page
* in chrome open the dev tools, right click on a data fetch request and block it
* refresh the page
* notice the failed network call, causes an error message

![screen shot 2018-05-22 at 11 38 25 am](https://user-images.githubusercontent.com/30807406/40373396-ce7442ec-5db4-11e8-8f3e-3935466edf15.png)


![screen shot 2018-05-22 at 11 39 04 am](https://user-images.githubusercontent.com/30807406/40373410-d33a0c6c-5db4-11e8-936f-87b560cec703.png)

![screen shot 2018-05-22 at 11 53 15 am](https://user-images.githubusercontent.com/30807406/40374406-023ceaa0-5db7-11e8-874b-569731325870.png)
